packages/grafana-ui/src/components/Toggletip/Toggletip.tsx

```diff
@@ -0,0 +1,301 @@
+  closeButton?: boolean;
+  /** Callback function to be called when the toggletip is closed */
+  onClose?: Function;
+  /** The preferred placement of the tooltip */
+  placement?: ToggletipPlacement;
+  /** The text or component that houses the content of the toggleltip */
+  content: ToggletipContent;
+  /** The text or component to be displayed on the toggletip's bottom */
+  footer?: JSX.Element | string;
+  /** The UI control users interact with to display toggletips */
+  children: JSX.Element;
+}
+
+export const Toggletip = React.memo(
+  ({
+    children,
+    theme = 'info',
+    placement = 'auto',
+    content,
+    title,
+    closeButton = true,
+    onClose,
+    footer,
+  }: ToggletipProps) => {
+    const styles = useStyles2(getStyles);
+    const containerStyle = styles[theme];
+    const contentRef = useRef(null);
+    const [controlledVisible, setControlledVisible] = React.useState(false);
+
+    useEffect(() => {
+      if (controlledVisible) {
+        const handleKeyDown = (enterKey: KeyboardEvent) => {
+          if (enterKey.key === 'Escape') {
+            setControlledVisible(false);
+            if (!!onClose) {
+              onClose();
+            }
+          }
+        };
+        document.addEventListener('keydown', handleKeyDown);
+        return () => {
+          document.removeEventListener('keydown', handleKeyDown);
+        };
+      } else if (controlledVisible === false && !!onClose) {
+        onClose();
+      }
+      return;
+    }, [controlledVisible, onClose]);
+
+    const { getArrowProps, getTooltipProps, setTooltipRef, setTriggerRef, visible, update } = usePopperTooltip({
+      visible: controlledVisible,
+      placement: placement,
+      interactive: true,
+      offset: [0, 8],
+      trigger: 'click',
+      onVisibleChange: setControlledVisible,
+    });
+
+    return (
+      <>
+        {React.cloneElement(children, {
+          ref: setTriggerRef,
+          tabIndex: 0,
+        })}
+        {visible && (
+          <Portal>
+            <div
+              data-testid="toggletip-content"
+              ref={setTooltipRef}
+              {...getTooltipProps({ className: containerStyle })}
+            >
+              {!!title && (
+                <div className="toggletip-header">
+                  {typeof title === 'string' && title}
+                  {title && React.isValidElement(title) && React.cloneElement(title)}
+                </div>
+              )}
+              {closeButton && (
+                <div data-testid="toggletip-header-close" className="toggletip-header-close">
+                  <IconButton
+                    aria-label="Close Toggletip"
+                    name="times"
+                    size="md"
+                    onClick={() => setControlledVisible(false)}
+                  />
+                </div>
+              )}
+              <div ref={contentRef} {...getArrowProps({ className: 'toggletip-arrow' })} />
```

Comment:
This should probably be something like:

```suggestion
              <div ref={contentRef} {...getArrowProps()} className={styles.toggleTipArrow} />
```

Reply:
This is how the library merges my styles and the internal styles of the arrow element. Take a look [here](https://github.com/mohsinulhaq/react-popper-tooltip#quick-start)